marlowe/test/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -419,6 +420,38 @@ choiceNotInBounds bounds =
     suchThat arbitrary $ \chosenNum -> not $ any (inBound chosenNum) bounds
 
 
+-- | Generate relevant Input content for a given input action
+-- interestingInput :: Bool -> Action -> [InputContent]
+-- interestingInput validity (Choice choiceId bounds) =
+--   map (IChoice choiceId) (interestingChoiceNum bounds)
+--   ++ map (\choiceId -> IChoice choiceId bounds) (interestingChoiceId choiceId)
+-- interestingInput validity (Choice choiceId bounds) = IChoice choiceId <$> interestingChoiceNum' validity bounds
+
+-- interestingChoiceId :: ChoiceId -> [ChoiceId]
+-- interestingChoiceId _ = error "not implemented yet"
+
+-- interestingParties :: Party -> [Party]
+-- interestingParties (Role "Alice") = [Role "Alice", Role "Alic", Role "Alices"]
+
+-- interestingChoiceNum :: [Bound] -> [ChosenNum]
+-- interestingChoiceNum bounds = concatMap interestingChoiceNum' True bounds
+
+interestingChoiceNum' :: Bool -> Bound -> [ChosenNum]
+interestingChoiceNum' True (Bound lower upper)  = validValues lower upper
+interestingChoiceNum' False (Bound lower upper) = invalidValues lower upper
+
+validValues :: Integer -> Integer -> [Integer]
+validValues lower upper = [x | x <- testValueRange, x >= lower  && x <= upper]
+
+invalidValues :: Integer -> Integer -> [Integer]
+invalidValues lower upper = [x | x <- testValueRange, x < lower  || x > upper]
+
+testValueRange :: [Integer]
+testValueRange = [(-rangeLimit)..rangeLimit]
+
+rangeLimit :: Integer
+rangeLimit = 10000
```

Comment:
Hi @bwbush! thanks to your suggestion of thinking about this problem as a number line, I think I was able to clean this code up a bit. However, it's still dependent on a hard-coded value in the `rangeLimit` function. We can also circle back on this when the tests start failing?

Reply:
Let's leave this as it is for right now. Later we'll use an `Arbitrary Integer` instance.